hazelcast/src/main/java/com/hazelcast/client/impl/spi/impl/ClientInvocationServiceImpl.java

```diff
@@ -294,6 +294,36 @@ void checkInvocationAllowed() throws IOException {
         connectionManager.checkInvocationAllowed();
     }
 
+    void checkUrgentInvocationAllowed(ClientInvocation invocation) {
+        if (!client.shouldCheckUrgentInvocations()) {
+            // If there were no Compact schemas to begin with, we don't need
+            // to perform the checks below. If the client didn't send a Compact
+            // schema up until this point, the retries or listener registrations
+            // could not send a schema, because if they were, we wouldn't hit
+            // this line.
+            return;
+        }
+
+        if (connectionManager.clientInitializedOnCluster()) {
```

Comment:
Yeah might save a check in the case we have some schemas and initialized on the cluster, which is the common case.

Reply:
Not necessarily, but it can very well be. As long as we do these two checks, the order between them doesn't really matter. Let me make it the first check